diff --git a/homebox.js b/homebox.js
index 75d31fb..f6d1748 100644
--- a/homebox.js
+++ b/homebox.js
@@ -274,30 +274,31 @@
         });
 
         // Add tooltips to icons
-        $portlet.find('.portlet-icon').tipsy({
-          gravity: 's',
-          title: function () {
-            switch ($(this).attr('class').replace('portlet-icon portlet-', '')) {
-            case 'close':
-              return Drupal.t('Close');
-            case 'maximize':
-              return Drupal.t('Maximize');
-            case 'minimize':
-              return Drupal.t('Minimize');
-            case 'minus':
-              return Drupal.t('Collapse');
-            case 'plus':
-              return Drupal.t('Expand');
-            case 'settings':
-              return Drupal.t('Settings');
-            }
-          }
-        });
-
-        // Remove tooltips on header clicks
-        $portletHeader.click(function () {
-          $('.tipsy').remove();
-        });
+        // Mukurtu patch -- disabling these tooltips because they prevent you from clicking on the buttons!
+        //$portlet.find('.portlet-icon').tipsy({
+        //  gravity: 's',
+        //  title: function () {
+        //    switch ($(this).attr('class').replace('portlet-icon portlet-', '')) {
+        //    case 'close':
+        //      return Drupal.t('Close');
+        //    case 'maximize':
+        //      return Drupal.t('Maximize');
+        //    case 'minimize':
+        //      return Drupal.t('Minimize');
+        //    case 'minus':
+        //      return Drupal.t('Collapse');
+        //    case 'plus':
+        //      return Drupal.t('Expand');
+        //    case 'settings':
+        //      return Drupal.t('Settings');
+        //    }
+        //  }
+        //});
+        //
+        //// Remove tooltips on header clicks
+        //$portletHeader.click(function () {
+        //  $('.tipsy').remove();
+        //});
       });
     }
   };
