Index: includes/media.fields.inc
===================================================================
--- includes/media.fields.inc	(revision 220)
+++ includes/media.fields.inc	(revision 233)
@@ -191,6 +191,8 @@
     $files = array();
     foreach ($items as $delta => $item) {
       $files[$item['fid']] = $item['file'];
+      $files[$item['fid']]->referencing_entity = $entity;
+      $files[$item['fid']]->referencing_field = $field['field_name'];
     }
 
     if (!empty($files)) {
