Index: file_entity_link/file_entity_link.module
===================================================================
--- file_entity_link/file_entity_link.module	(revision 231)
+++ file_entity_link/file_entity_link.module	(working copy)
@@ -32,18 +32,27 @@
  * @param unknown_type $view_mode
  * @param unknown_type $langcode
  */
-function file_entity_link_file_view($file, $view_mode, $langcode){
+function file_entity_link_file_view_alter(&$build, $type){
   $settings = variable_get(FILE_ENTITY_LINK_SETTINGS_VAR, array());
-  if(isset($settings[$file->type][$view_mode]) && $settings[$file->type][$view_mode] && !empty($file->referencing_entity)){
-    if(isset($file->referencing_entity->nid)){
-      $uri = entity_uri('node',$file->referencing_entity);
+  $file_type = $build['#file']->type;
+  $view_mode = $build['#view_mode'];
+  if(isset($settings[$file_type][$view_mode]) && $settings[$file_type][$view_mode] && !empty($build['#file']->referencing_entity)){
+    switch ($settings[$file_type][$view_mode]) {
+      case 1:
+        if (isset($build['#file']->referencing_entity->nid)) {
+          $uri = entity_uri('node', $build['#file']->referencing_entity);
+        }
+        elseif (isset($build['#file']->referencing_entity->tid)) {
+          $uri = entity_uri('taxonomy_term', $build['#file']->referencing_entity);
+        }
+        break;
+      case 2:
+        $uri = array('path' => 'media/' . $build['#file']->fid, 'options' => array());
+        break;
     }
-    elseif(isset($file->referencing_entity->tid)){
-      $uri = entity_uri('taxonomy_term',$file->referencing_entity);
-    }
     if(!empty($uri)){
-      $file->content['file'] = array(
-     	'#contents' => $file->content['file'],
+      $build['file'] = array(
+     	'#contents' => $build['file'],
         '#uri' => $uri,
         '#theme' => 'file_entity_link_add_link'
       );
@@ -65,8 +74,8 @@
       '#type' => 'item',
       '#title' => 'Link options',
       'link_file' => array(
-        '#type' => 'checkbox',
-        '#title' => 'Link file to referencing entity',
+        '#type' => 'select',
+        '#options' => array('0' => 'No link', '1' => 'Link file to referencing entity', '2' => 'Link file to media page'),
         '#default_value' => (isset($settings[$form['#file_type']][$form['#view_mode']]) ? $settings[$form['#file_type']][$form['#view_mode']] : FALSE),
       )
     );
