diff --git a/community_tags.module b/community_tags.module
index aad8c6a..ccb090b 100644
--- a/community_tags.module
+++ b/community_tags.module
@@ -239,18 +239,19 @@ function community_tags_taxonomy($op = NULL, $type = NULL, $term = NULL) {
  * Implements hook_user_cancel().
  */
 function community_tags_user_cancel($edit, $account, $method) {
-  // if user is deleted then remove all ctags for the user.
-  // @todo consider option of moving all tags to a "dead" user so tags are not lost
-  // maybe something to add with anonymous user support if method is appropriate.
-  _community_tags_user_delete($user);
+  // If user is deleted then remove all ctags for the user.
+  // @todo consider option of moving all tags to a "dead" user so tags
+  // are not lost maybe something to add with anonymous user support if
+  // method is appropriate.
+  _community_tags_user_delete($account);
 }
 
 /**
  * Implements hook_user_delete().
  */
 function community_tags_user_delete($account) {
-  // if user is deleted then remove all ctags for the user.
-  _community_tags_user_delete($user);
+  // If user is deleted then remove all ctags for the user.
+  _community_tags_user_delete($account);
 }
 
 /**
@@ -893,8 +894,8 @@ function _community_tags_add_mixed_terms_to_node(&$node, $terms_to_add) {
 function _community_tags_add_terms_to_node(&$node, $tids_to_add, $vid) {
   $count_of_terms_added = 0;
 
-  // add to all fields that don't have it
-  foreach($terms_to_add as $tid) {
+  // Add to all fields that don't have it.
+  foreach($tids_to_add as $tid) {
     $count_of_terms_added += _community_tags_add_term_to_node($node, $tid, $vid);
   }
 
