diff --git a/includes/common.inc b/includes/common.inc
index a32930a5..0c7fb889 100644
--- a/includes/common.inc
+++ b/includes/common.inc
@@ -7638,7 +7638,7 @@ function drupal_implode_tags($tags) {
 
     $encoded_tags[] = $tag;
   }
-  return implode(', ', $encoded_tags);
+  return implode('; ', $encoded_tags);
 }
 
 /**
diff --git a/misc/autocomplete.js b/misc/autocomplete.js
index af090713..b2cc5253 100644
--- a/misc/autocomplete.js
+++ b/misc/autocomplete.js
@@ -277,7 +277,7 @@ Drupal.ACDB.prototype.search = function (searchString) {
   // Skip empty search strings, or search strings ending with a comma, since
   // that is the separator between search terms.
   if (searchString.length <= 0 ||
-    searchString.charAt(searchString.length - 1) == ',') {
+    searchString.charAt(searchString.length - 1) == ';') {
     return;
   }
 
diff --git a/modules/taxonomy/taxonomy.pages.inc b/modules/taxonomy/taxonomy.pages.inc
index 01919878..f4b99717 100644
--- a/modules/taxonomy/taxonomy.pages.inc
+++ b/modules/taxonomy/taxonomy.pages.inc
@@ -165,12 +165,12 @@ function taxonomy_autocomplete($field_name = '', $tags_typed = '') {
       ->execute()
       ->fetchAllKeyed();
 
-    $prefix = count($tags_typed) ? drupal_implode_tags($tags_typed) . ', ' : '';
+    $prefix = count($tags_typed) ? drupal_implode_tags($tags_typed) . '; ' : '';
 
     foreach ($tags_return as $tid => $name) {
       $n = $name;
       // Term names containing commas or quotes must be wrapped in quotes.
-      if (strpos($name, ',') !== FALSE || strpos($name, '"') !== FALSE) {
+      if (strpos($name, ';') !== FALSE || strpos($name, '"') !== FALSE) {
         $n = '"' . str_replace('"', '""', $name) . '"';
       }
       $term_matches[$prefix . $n] = check_plain($name);
