From a6168444687b40e502488a15cd665f74e3d59343 Mon Sep 17 00:00:00 2001
From: Steve Taylor <steve_taylor@wsu.edu>
Date: Fri, 15 Apr 2016 09:35:39 -0700
Subject: [PATCH] Stop double quoting quotes in term autocomplete

---
 modules/taxonomy/taxonomy.pages.inc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modules/taxonomy/taxonomy.pages.inc b/modules/taxonomy/taxonomy.pages.inc
index b4dd3fd..51d1943 100644
--- a/modules/taxonomy/taxonomy.pages.inc
+++ b/modules/taxonomy/taxonomy.pages.inc
@@ -170,9 +170,11 @@ function taxonomy_autocomplete($field_name = '', $tags_typed = '') {
     foreach ($tags_return as $tid => $name) {
       $n = $name;
       // Term names containing commas or quotes must be wrapped in quotes.
+      /*
       if (strpos($name, ';') !== FALSE || strpos($name, '"') !== FALSE) {
         $n = '"' . str_replace('"', '""', $name) . '"';
       }
+      */
       $term_matches[$prefix . $n] = check_plain($name);
     }
   }
-- 
2.1.4

