diff --git a/includes/batch.inc b/includes/batch.inc
index c98135beb..e26ccb3b3 100644
--- a/includes/batch.inc
+++ b/includes/batch.inc
@@ -466,6 +466,10 @@ function _batch_finished() {
     }
 
     // Use drupal_redirect_form() to handle the redirection logic.
+    if (isset($_batch['form_state']['redirect']['path'])) {
+      array_unshift($_batch['form_state']['redirect'], $_batch['form_state']['redirect']['path']);
+      unset($_batch['form_state']['redirect']['path']);
+    }
     drupal_redirect_form($_batch['form_state']);
 
     // If no redirection happened, redirect to the originating page. In case the
