diff --git a/modules/system/system.admin.inc b/modules/system/system.admin.inc
index 22c202c..4ff95d7 100644
--- a/modules/system/system.admin.inc
+++ b/modules/system/system.admin.inc
@@ -706,17 +706,23 @@ function system_theme_settings_submit($form, &$form_state) {
   if (!empty($values['logo_upload'])) {
     $file = $values['logo_upload'];
     unset($values['logo_upload']);
-    $filename = file_unmanaged_copy($file->uri);
+    $path = 'public://' . $file->filename;
+    $file->destination = $path;
+    $file->status = 1;
+    $file = file_copy($file, $file->destination);
     $values['default_logo'] = 0;
-    $values['logo_path'] = $filename;
+    $values['logo_path'] = $file->uri;;
     $values['toggle_logo'] = 1;
   }
   if (!empty($values['favicon_upload'])) {
     $file = $values['favicon_upload'];
     unset($values['favicon_upload']);
-    $filename = file_unmanaged_copy($file->uri);
+    $path = 'public://' . $file->filename;
+    $file->destination = $path;
+    $file->status = 1;
+    $file = file_copy($file, $file->destination);
     $values['default_favicon'] = 0;
-    $values['favicon_path'] = $filename;
+    $values['favicon_path'] = $file->uri;
     $values['toggle_favicon'] = 1;
   }
 
