diff --git a/og.test b/og.test
index c267b8d..44fa3d0 100644
--- a/og.test
+++ b/og.test
@@ -1024,7 +1024,7 @@ class OgMigrate7000TestCase extends UpgradePathTestCase {
 /**
  * Test the complex widget.
  */
-class OgComplexWidgetTestCase extends DrupalWebTestCase {
+class OgFieldWidgetTestCase extends DrupalWebTestCase {
 
   public static function getInfo() {
     return array(
@@ -1094,6 +1094,24 @@ class OgComplexWidgetTestCase extends DrupalWebTestCase {
   }
 
   /**
+   * Test "field widget settings" of the OG reference widget.
+   */
+  function testFieldSettings() {
+    $user = $this->drupalCreateUser(array('administer content types', 'administer group', 'access content', ));
+    $this->drupalLogin($user);
+
+    $widgets = field_info_widget_types();
+    foreach($widgets as $widget => $settings){
+      if(in_array('entityreference', $settings['field types'])){
+        $widget_type_url = "admin/structure/types/manage/post/fields/".OG_AUDIENCE_FIELD."/widget-type";
+        $this->drupalPost($widget_type_url, array('widget_type' => $widget), 'Continue');
+        $edit_url = "admin/structure/types/manage/post/fields/".OG_AUDIENCE_FIELD;
+        $this->drupalPost($edit_url, array(), 'Save settings');
+      }
+    }
+  }
+
+  /**
    * Test non-accessible group IDs are saved, upon form submit.
    */
   function testHiddenGroupIds() {
diff --git a/plugins/entityreference/behavior/OgBehaviorHandler.class.php b/plugins/entityreference/behavior/OgBehaviorHandler.class.php
index 7191a68..cc3db64 100644
--- a/plugins/entityreference/behavior/OgBehaviorHandler.class.php
+++ b/plugins/entityreference/behavior/OgBehaviorHandler.class.php
@@ -260,7 +260,9 @@ class OgBehaviorHandler extends EntityReference_BehaviorHandler_Abstract {
     }
     else {
       foreach ($items as $item) {
-        $values['default'][] = $item['target_id'];
+        if (!entityreference_field_is_empty($item, $field) && $item['target_id'] !== NULL) {
+          $values['default'][] = $item['target_id'];
+        }
       }
     }
 
