diff --git a/og.module b/og.module
index dac0ab7..8a0e5cd 100755
--- a/og.module
+++ b/og.module
@@ -2240,6 +2240,12 @@ function og_user_access_entity($perm, $entity_type, $entity, $account = NULL) {
       return TRUE;
     }
     else {
+      // PATCH BY SHIRAZ / MUKURTU
+      // The following is buggy. Group admins of Community have edit permission on Community nodes, so they are then falsely given edit permission on CP nodes (a subgroup) therein that they don't own. See: http://drupal.aspcode.net/ppst/63547274810018958098124/debuging-og-access-field
+      // So simply returning FALSE here instead.
+      return FALSE;
+      // END PATCH
+
       // An entity can be a group and group content in the same time. The group
       // didn't return TRUE, but the user still might have access to the
       // permission in group content context.
