Index: media_browser_plus.install
===================================================================
--- media_browser_plus.install	(revision 222)
+++ media_browser_plus.install	(working copy)
@@ -150,10 +150,10 @@
       'operations' => array(
         array('media_browser_plus_folder_media_import', array($term->tid, $dir)),
       ),
-      'finished' => 'media_browser_plus_folder_media_import_finished',
       'file' => drupal_get_path('module', 'media_browser_plus') . '/includes/media_browser_plus.admin.inc',
     );
     batch_set($batch);
+    variable_del('media_folder_import_start');
   }
 }
 
