diff --git a/geofield.feeds.inc b/geofield.feeds.inc
index a53a315..db9cd5a 100644
--- a/geofield.feeds.inc
+++ b/geofield.feeds.inc
@@ -142,6 +142,21 @@ function geofield_set_target_simple($source, $entity, $target, $values) {
     $delta++;
   }
 
+  foreach ($field[LANGUAGE_NONE] as $delta => $value) {
+    if (!empty($value['lat']) && !empty($value['lon'])) {
+      // Build up geom data.
+      $field[LANGUAGE_NONE][$delta] = geofield_compute_values($value, 'latlon');
+    }
+    elseif (!empty($value['top']) &&
+            !empty($value['right']) &&
+            !empty($value['bottom']) &&
+            !empty($value['left'])) {
+      // Set format and geom data for geofield_bounds
+      $field[LANGUAGE_NONE][$delta]['geom'] = $field[LANGUAGE_NONE][$delta];
+      $field[LANGUAGE_NONE][$delta]['input_format'] = GEOFIELD_INPUT_BOUNDS;
+    }
+  }
+
   $entity->$field_name = $field;
 }
 
