diff --git a/modules/fields/atom_reference/atom_reference.module b/modules/fields/atom_reference/atom_reference.module
index 27effec..58c78ab 100644
--- a/modules/fields/atom_reference/atom_reference.module
+++ b/modules/fields/atom_reference/atom_reference.module
@@ -163,13 +163,20 @@ function atom_reference_field_formatter_settings_summary($field, $instance, $vie
   return $summary;
 }
 
+
 /**
  * Implements hook_field_formatter_view.
  */
+// Patched by Shiraz -- in search results view:
+// 1. Return the first *image* in a scald node, falling back to first video or audio if no image.
+// 2. Show thumbnails instead of player for audio and video files if exists, otherwise use our default thumbnails according to atom type
 function atom_reference_field_formatter_view($entity_type, $entity, $field, $instance, $langcode, $items, $display) {
+
   $render_context = $display['type'];
   $contexts = scald_contexts();
   $element = array();
+  $priority_elements = array();
+  $regular_elements = array();
 
   $uri = false;
 
@@ -181,15 +188,54 @@ function atom_reference_field_formatter_view($entity_type, $entity, $field, $ins
   if (!empty($contexts[$render_context])) {
     foreach ($items as $delta => $item) {
       $sid = $item['sid'];
+      $atom = scald_is_registered($sid);
       $options = array();
       if ($uri) {
         $options['link'] = $uri['path'];
       }
-      $element[$delta] = array('#markup' => scald_render($sid, $render_context, drupal_json_encode($options)));
+
+      if ($display['type'] <> 'ma_scald_search_results' && $display['type'] <> 'mukurtu_scald_related_items') {
+        $priority_elements[$delta] = array('#markup' => scald_render($sid, $render_context, drupal_json_encode($options)));
+      }
+
+      else {
+
+        // Scald Atom is Image
+        if ($atom->type == 'image') {
+          $priority_elements[] = array('#markup' => scald_render($sid, $render_context, drupal_json_encode($options)));
+        }
+
+        // Scald Atom is not an image
+        else {
+
+          // Use the atom thumbnail if there is one
+          if (isset($atom->scald_thumbnail[LANGUAGE_NONE][0]['uri'])) {
+            $image_styles = array (
+              'ma_scald_search_results' => 'medium',
+              'mukurtu_scald_related_items' => 'related_items_teaser'
+            );
+            $thumbnail_image = theme_image_style (array (
+              'style_name' => $image_styles[$display['type']],
+              'path' => $atom->scald_thumbnail[LANGUAGE_NONE][0]['uri'],
+            ));
+          }
+          // If no atom thumbnail, use our default thumbnails
+          else {
+            $image_widths = array (
+              'ma_scald_search_results' => '220',
+              'mukurtu_scald_related_items' => '280'
+            );
+            $thumbnail_image = '<img src = "/' . drupal_get_path('module', 'ma_digitalheritage') . '/images/default_' . $atom->type . '_thumbnail_' . $image_widths[$display['type']] . '.jpg">';
+          }
+          $regular_elements[] = array('#markup' => $thumbnail_image);
+        }
+      }
     }
   }
 
+  $element = array_merge($priority_elements, $regular_elements);
   return $element;
+
 }
 
 /**
