diff --git a/modules/providers/scald_youtube/scald_youtube.module b/modules/providers/scald_youtube/scald_youtube.module
index 38b190a..b366dfb 100644
--- a/modules/providers/scald_youtube/scald_youtube.module
+++ b/modules/providers/scald_youtube/scald_youtube.module
@@ -109,12 +109,23 @@ function scald_youtube_scald_fetch($atom, $type) {
  */
 function scald_youtube_scald_prerender($atom, $context, $options, $mode) {
   if ($mode == 'atom') {
+    //load context configuration to retrieve dimension data if present
+    $context_config = scald_context_config_load($context);
+    $video_width = (isset($atom->data['video_width'])) ? $atom->data['video_width'] : 480;
+    $video_height = (isset($atom->data['video_height'])) ? $atom->data['video_height'] : 365;
+    //allow context configuration to override video dimension variables
+    if(isset($context_config->data['width'])){
+      $video_width = $context_config->data['width'];
+    }
+    if(isset($context_config->data['height'])){
+      $video_height = $context_config->data['height'];
+    }
     $atom->rendered->player = theme('scald_youtube_player',
       array('vars' =>
         array(
           'video_id'      => $atom->base_id,
-          'video_width'   => ($atom->data['video_width'] ? $atom->data['video_width'] : 480),
-          'video_height'  => ($atom->data['video_height'] ? $atom->data['video_height'] : 365),
+          'video_width'   => $video_width,
+          'video_height'  => $video_height,
           'thumbnail'     => $atom->rendered->thumbnail_source_url,
         ),
         'atom' => $atom,
