diff --git a/includes/ajax.inc b/includes/ajax.inc
index 90e4ca9..9e6e6ad 100644
--- a/includes/ajax.inc
+++ b/includes/ajax.inc
@@ -361,13 +361,13 @@ function views_ajax_autocomplete_taxonomy($vid, $tags_typed = '') {
       ->execute()
       ->fetchAllKeyed();

-    $prefix = count($tags_typed) ? drupal_implode_tags($tags_typed) . ', ' : '';
+    $prefix = count($tags_typed) ? drupal_implode_tags($tags_typed) . '; ' : '';

     $term_matches = array();
     foreach ($tags_return as $tid => $name) {
       $n = $name;
       // Term names containing commas or quotes must be wrapped in quotes.
-      if (strpos($name, ',') !== FALSE || strpos($name, '"') !== FALSE) {
+      if (strpos($name, ';') !== FALSE || strpos($name, '"') !== FALSE) {
         $n = '"' . str_replace('"', '""', $name) . '"';
       }
       // Add term name to list of matches.
