diff --git a/modules/providers/scald_audio/scald_audio_html5.tpl.php b/modules/providers/scald_audio/scald_audio_html5.tpl.php
index 274b854..bcf5527 100644
--- a/modules/providers/scald_audio/scald_audio_html5.tpl.php
+++ b/modules/providers/scald_audio/scald_audio_html5.tpl.php
@@ -4,8 +4,13 @@
  * Scald Audio player template
  * Changed to use HTML5 Audio element by Mark Conroy - https://www.drupal.org/user/336910
  */
+
+$type = "";
+if(substr($vars['audio_uri'], -3, 3) == 'm4a') {
+  $type = 'type="audio/mp4"';
+  }
 ?>
 <audio controls preload="metadata">
- <source src="<?php echo file_create_url($vars['audio_uri']) ?>">
+ <source src="<?php echo file_create_url($vars['audio_uri']) ?>" <?php echo $type ?>>
  <p>Your browser does not support the audio element.</p>
 </audio>
