diff --git a/partial_date.admin.inc b/partial_date.admin.inc
index 6a56311..0c3be46 100644
--- a/partial_date.admin.inc
+++ b/partial_date.admin.inc
@@ -18,15 +18,20 @@ function _partial_date_field_info() {
       // Includes each component and any, any_form, and any_to.
       'minimum_components' => array(),
     ),
+    'property_callbacks' => array(
+      'partial_date_property_info_callback',
+    ),
   );
   return array(
     'partial_date' => $base + array(
       'label' => t('Partial date and time'),
       'description' => t('This field stores and renders partial dates.'),
+      'property_type' => 'partial_date',
     ),
     'partial_date_range' => $base + array(
       'label' => t('Partial date and time range'),
       'description' => t('This field stores and renders partial dates.'),
+      'property_type' => 'partial_date_range',
     ),
   );
 }
diff --git a/partial_date.module b/partial_date.module
index dcd9069..638de0c 100644
--- a/partial_date.module
+++ b/partial_date.module
@@ -110,6 +110,115 @@ function partial_date_field_info() {
 }
 
 /**
+ * Property info callback function.
+ */
+function partial_date_property_info_callback(&$info, $entity_type, $field, $instance, $field_type) {
+  $property = &$info[$entity_type]['bundles'][$instance['bundle']]['properties'][$field['field_name']];
+
+  $property['getter callback'] = 'entity_metadata_field_verbatim_get';
+  $property['setter callback'] = 'entity_metadata_field_verbatim_set';
+  unset($property['query callback']);
+
+  $property['property info']['timestamp'] = array(
+    'type' => 'integer',
+    'label' => t('Timestamp'),
+    'setter callback' => 'entity_property_verbatim_set',
+  );
+
+  $property['property info']['txt_short'] = array(
+    'type' => 'text',
+    'label' => t('Short description of the date'),
+    'setter callback' => 'entity_property_verbatim_set',
+  );
+
+  $property['property info']['txt_long'] = array(
+    'type' => 'text',
+    'label' => t('Long description of the date'),
+    'setter callback' => 'entity_property_verbatim_set',
+  );
+
+  $property['property info']['year'] = array(
+    'type' => 'integer',
+    'label' => t('Year'),
+    'setter callback' => 'entity_property_verbatim_set',
+  );
+
+  $property['property info']['month'] = array(
+    'type' => 'integer',
+    'label' => t('Month'),
+    'setter callback' => 'entity_property_verbatim_set',
+  );
+
+  $property['property info']['day'] = array(
+    'type' => 'integer',
+    'label' => t('Day'),
+    'setter callback' => 'entity_property_verbatim_set',
+  );
+
+  $property['property info']['hour'] = array(
+    'type' => 'integer',
+    'label' => t('Hour'),
+    'setter callback' => 'entity_property_verbatim_set',
+  );
+
+  $property['property info']['minute'] = array(
+    'type' => 'integer',
+    'label' => t('Minute'),
+    'setter callback' => 'entity_property_verbatim_set',
+  );
+
+  $property['property info']['second'] = array(
+    'type' => 'integer',
+    'label' => t('Second'),
+    'setter callback' => 'entity_property_verbatim_set',
+  );
+
+  if ($field_type['property_type'] == 'partial_date_range') {
+    $property['property info']['timestamp_to'] = array(
+      'type' => 'integer',
+      'label' => t('Timestamp (to)'),
+      'setter callback' => 'entity_property_verbatim_set',
+    );
+
+    $property['property info']['year_to'] = array(
+      'type' => 'integer',
+      'label' => t('Year (to)'),
+      'setter callback' => 'entity_property_verbatim_set',
+    );
+
+    $property['property info']['month_to'] = array(
+      'type' => 'integer',
+      'label' => t('Month (to)'),
+      'setter callback' => 'entity_property_verbatim_set',
+    );
+
+    $property['property info']['day_to'] = array(
+      'type' => 'integer',
+      'label' => t('Day (to)'),
+      'setter callback' => 'entity_property_verbatim_set',
+    );
+
+    $property['property info']['hour_to'] = array(
+      'type' => 'integer',
+      'label' => t('Hour (to)'),
+      'setter callback' => 'entity_property_verbatim_set',
+    );
+
+    $property['property info']['minute_to'] = array(
+      'type' => 'integer',
+      'label' => t('Minute (to)'),
+      'setter callback' => 'entity_property_verbatim_set',
+    );
+
+    $property['property info']['second_to'] = array(
+      'type' => 'integer',
+      'label' => t('Second (to)'),
+      'setter callback' => 'entity_property_verbatim_set',
+    );
+  }
+}
+
+/**
  * Implements hook_field_settings_form().
  */
 function partial_date_field_settings_form($field, $instance, $has_data) {
